var/spack/repos/builtin/packages/openspeedshop-utils/package.py

```diff
@@ -0,0 +1,345 @@
+
+    depends_on("cbtf-argonavis@develop", when='@develop+cuda')
+    depends_on("cbtf-argonavis@1.9.1.0", when='@2.3.1.3+cuda')
+
+    # For MRNet
+    depends_on("mrnet@5.0.1-3:+cti", when='@develop+cti')
+    depends_on("mrnet@5.0.1-3:+lwthreads", when='@develop')
+
+    depends_on("mrnet@5.0.1-3:+cti", when='@2.3.1.3+cti')
+    depends_on("mrnet@5.0.1-3:+lwthreads", when='@2.3.1.3')
+
+    parallel = False
+
+    build_directory = 'build_openspeedshop'
+
+    def set_CrayLoginNode_cmakeOptions(self, spec, cmakeOptions):
+        # Appends to cmakeOptions the options that will enable the appropriate
+        # Cray login node libraries
+
+        CrayLoginNodeOptions = []
+        rt_platform = "cray"
+
+        # How do we get the compute node (CNL) cbtf package install
+        # directory path?
+        # spec['cbtf'].prefix is the login node value for this build, as
+        # we only get here when building the login node components and
+        # that is all that is known to spack.
+        be_ck = spack.store.db.query_one('cbtf-krell arch=cray-CNL-haswell')
+
+        # Equivalent to install-tool cmake arg:
+        # '-DCBTF_KRELL_CN_RUNTIME_DIR=%s'
+        #               % <base dir>/cbtf_v2.3.1.release/compute)
+        CrayLoginNodeOptions.append('-DCBTF_KRELL_CN_RUNTIME_DIR=%s'
+                                    % be_ck.prefix)
+        CrayLoginNodeOptions.append('-DRUNTIME_PLATFORM=%s'
+                                    % rt_platform)
+
+        cmakeOptions.extend(CrayLoginNodeOptions)
+
+    def cmake_args(self):
+
+        spec = self.spec
+
+        compile_flags = "-O2 -g"
+
+        cmake_args = []
+
+        # Indicate building cbtf vers (transfer rawdata files)
+        instrumentor_setting = "cbtf"
+
+        if spec.satisfies('+runtime'):
+            # Appends base options to cmake_args
+            self.set_defaultbase_cmakeOptions(spec, cmake_args)
+
+            cmake_args.extend(
+                ['-DCMAKE_CXX_FLAGS=%s'  % compile_flags,
+                 '-DCMAKE_C_FLAGS=%s'    % compile_flags,
+                 '-DINSTRUMENTOR=%s' % instrumentor_setting,
+                 '-DCBTF_DIR=%s' % spec['cbtf'].prefix,
+                 '-DCBTF_KRELL_DIR=%s' % spec['cbtf-krell'].prefix,
+                 '-DMRNET_DIR=%s' % spec['mrnet'].prefix])
+
+        else:
+
+            # Appends base options to cmake_args
+            self.set_defaultbase_cmakeOptions(spec, cmake_args)
+            cmake_args.extend(
+                ['-DCMAKE_CXX_FLAGS=%s' % compile_flags,
+                 '-DCMAKE_C_FLAGS=%s' % compile_flags,
+                 '-DINSTRUMENTOR=%s' % instrumentor_setting,
+                 '-DSQLITE3_DIR=%s' % spec['sqlite'].prefix,
+                 '-DCBTF_DIR=%s' % spec['cbtf'].prefix,
+                 '-DCBTF_KRELL_DIR=%s' % spec['cbtf-krell'].prefix,
+                 '-DMRNET_DIR=%s' % spec['mrnet'].prefix])
+
+            cmake_args.extend(['-DBUILD_QT3_GUI=FALSE'])
```

Comment:
Is the idea that `+runtime` builds a private instance of qt 3? Or is it that one will always be building qt 4/5 when `+runtime` is set?